packages/web-frontend/src/screens/LandingPage/MultiProjectLandingPage.js

```diff
@@ -81,13 +79,21 @@ export function MultiProjectLandingPage() {
           <ProjectCardList
             projects={projects}
             actions={{
-              [PROJECT_ACCESS_TYPES.ALLOWED]: navigateToProject,
-              [PROJECT_ACCESS_TYPES.PENDING]: navigateToRequestProjectAccess,
-              [PROJECT_ACCESS_TYPES.DENIED]: isUserLoggedIn
-                ? navigateToRequestProjectAccess
-                : navigateToLogin,
+              [PROJECT_ACCESS_TYPES.ALLOWED]: ({ project }) => (
+                <ProjectAllowedButton onClick={() => navigateToProject(project)} />
+              ),
+              [PROJECT_ACCESS_TYPES.PENDING]: ({ project }) => (
+                <ProjectPendingButton onClick={() => navigateToRequestProjectAccess(project)} />
+              ),
+              [PROJECT_ACCESS_TYPES.DENIED]: ({ project }) => {
+                if (isUserLoggedIn) {
+                  return (
+                    <ProjectDeniedButton onClick={() => navigateToRequestProjectAccess(project)} />
+                  );
+                }
+                return <ProjectLoginButton onClick={navigateToLogin} />;
+              },
```

Comment:
I wonder if there is a way to simplify this `actions` logic? It looks like the only difference between buttons in the views that use it are the onClick actions themselves, and the landing pages has a `Log in` button when permission is denied because the user is not logged in. Could we perhaps keep it with the actions as they were and potentially have a flag for `useLoginButton` or something in the `ProjectCardList` component? I know that's not ideal either, so feel free to push back.

Reply:
They are quite similar but I was anticipating the upcoming ticket for changing the cards on the main landing page where I think the cards will end being exactly the same or they might diverge a bit so I liked the flexibility. Either was I'm keen to leave as is for now if that's ok.